### ISSUE TITLE
Ships with 'Mk.' in name not said correctly

### ISSUE THREAD

**Issue by Darkcyde13** [cmdrmcdonald/EliteDangerousDataProvider#87]
First, I apologise for the length of this, I wanted to test as much as possible to ensure what I was saying is reproduceable and accurate.

In Ship Purchased {event.ship} and in Bounty Awarded {event.target}, the speech responder reads the 'full stop' punctuation in the middle of the ship model (Mk.) as an end of sentence, leaving a long pause in the middle of the name.  It also says 'Mk' rather than 'Mark'.

For example, when purchasing a 'Cobra Mk. III', it is said as 'Cobra MK [pause] I I I' and not as 'Cobra Mark Three'.  Same goes for 'Viper Mk. III' e.t.c.

When selling, or swapping to/from, a 'Cobra Mk. III' the speech says it correctly as 'Cobra Mark Three'.  However, if {event.ship} is used on it's own in 'Ship Sold' (rather than as part of ShipName), it also does not say it correctly.  I am assuming that 'ShipName()' is getting the model (if there is no player given name) from somewhere else as it seems to say these perfectly ok.

Also, when swapping ships, if I swap out from (store) a Mk.IV model, the word 'four' sounds really weird, like 'flwer' or something.  If I swap between a Cobra Mk. IV and a Viper Mk. IV, back and fourth, the ship going into storage sounds weird when saying the 'four', but the ship comming out of storage says 'four' fine. Could there be something wrong with the event.storedshipid variable?

It would seem that ShipDetails([variable]).model (and possibly event.ship) simply says whatever is in the [variable] part as I've even tried ShipDetails with "cobra mk. xx" and it says 'Cobra MK [pause] Twenty' which obviously doesn't exist in game.  I can understand this, as it means much less programming to achieve the desired result.  However, I tried it with 'cobra mk. v' and that just says the 'v' as the letter vee, not the number five.

I also noted that if you enclose event.ship or ShipDetails within {P()} like in the default Ship Purchased script, for a Mk. III or Mk. VI model, it will say the Roman numerals as the individual letters rather than the decimal number. i.e. {P(ShipDetails("cobra mk. iii").model)} comes out as 'Cobra MK [pause] [eye-eye-eye]'. However, it does seem to say 'viii' (eight), 'ix' (nine) and 'xx' (twenty) perfectly fine still, so it's inconsistent with the conversion from Roman to decimal.

After running a couple of tests with the write text copies to file, I see that the passed variables are capitalised (so IV not iv). When using capitals for the numbers, in {event.ship} etc, they are all separated by a space and thus said as letters.  When used in ShipName, they are not separated and so are said as numbers.

I found that lower case is hit and miss if it's said as a number or the letters.  For example, 'iv'. 'v', or 'vi' is said as the letters, but 'vii' and 'viii' are said as numbers seven and eight.  Not sure if this is anything you can correct?

TL;DR
In 'Ship Purchased', 'Ship Swapped', and 'Ship Sold':-
  {event.ship} reads the full stop as an end of sentence, adding a pause in the middle of the ship model name.
  ShipName() (in Ship Swapped and Ship Sold) says the model correctly ('Mark Three' vs 'MK [pause] Three)
  Roman numerals are said as letters in 'Ship Purchased', because of {P()}.

In 'Bounty Awarded':-
  {event.target} does the same as above, reading 'M K [pause] I I I'.

Roman numerals are inconsistantly converted to decimal, sometimes left just as the letters.

This is all using EDDI beta 3 and E:D v2.2.03 beta 4.

Regards,
Darkcyde

P.S.  I hope you don't think I'm just picking holes in your software.  I love what EDDI does and am very grateful to you for writing it.  I'm just a bit of a perfectionist sometimes. ;)

**Comment by cmdrmcdonald** [comment]
The pronunciation of names is down to the particular TTS voice.  EDDI does provide ships with phonetic version of the name to provide accurate representations, but it's generally locked away behind a function.  I haven't tried it but it's possible that

    ShipDetails('Cobra Mk. III').SpokenModel()

would give the correct value.  That said, it really should be picked up by P(), which I'll take a look at.

**Comment by Darkcyde13** [comment]
I tried the following in a new script and tried all the voices I have, which are the three default Win10 voices, plus Ivona Amy British English.  They all gave the same results as follows:-

{ShipDetails("Cobra Mk. III").SpokenModel()}.
Nothing spoken, so I guess this doesn't work, but there were no errors either.

{P(ShipDetails("Cobra Mk. III").model))}.
This says "Cobra MK [pause] [eye-eye-eye]"

{ShipDetails("Cobra Mk. III").model)}.
This says "Cobra MK [pause] Three"

If it's not possible to fix, I can work around it by creating a new function with the following line for each of the ships with 'Mk.' in the name:-
{if shipname = "Cobra Mk. III": {set shipname to "Cobra Mark Three"}}
As it's only the ship name that's causing the difficulties, this won't be too much of a hassle to use.

**Comment by cmdrmcdonald** [comment]
P() has been updated to include ship models in its translation list so {P(ship.model)} will work in the next release.
